hathor/p2p/manager.py

```diff
@@ -274,7 +300,11 @@ def on_peer_disconnect(self, protocol: HathorProtocol) -> None:
                 # chance it can happen if both connections start at the same time and none of them has
                 # reached READY state while the other is on PEER_ID state
                 self.connected_peers[protocol.peer.id] = existing_protocol
-        self.pubsub.publish(HathorEvents.NETWORK_PEER_DISCONNECTED, protocol=protocol)
+        self.pubsub.publish(
```

Comment:
I'm struggling to understand why your changes have affected the simulation of the soft voided tests. Do you understand it?

Reply:
I've tried debugging this to find differences when executing the tests with and without my changes.

I know which of my changes specifically triggers the problem. It's the one in line `240` of this file. If I remove it, the tests work without modifications.

I got down to the point where I noticed that the tx value generated in `hathor/simulator/tx_generator.py:99` was the difference that impacted the tx hashes. After 16 generated transactions, this value starts to be different when comparing the test runs with and without my changes.

Initially I thought that my changes could be impacting the random number generator somehow, but I don't think this is the case anymore, because actually the `balance.available` there is also coming different in the 17th generated transaction when comparing the test runs.

So maybe there is some kind of race condition that got impacted because I have added a new `self.clock.callLater` call, which is impacting the wallet balance during the tests. But I didn't went farther than this.

Reply:
Got it. Thanks for the explanation.